cmd/jujud-controller/main_debug.go

```diff
@@ -0,0 +1,18 @@
+// Copyright 2012, 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+//go:build debug
+
+package main
+
+import (
+	"github.com/juju/juju/internal/dlw"
+	"github.com/juju/juju/internal/dlw/config"
+)
+
+func init() {
+	Main = dlw.Wrap(config.Default(),
+		config.WithPort(10123),
```

Comment:
At the moment I feel uncomfortable placing this in the code base until we can get some definitions around how we expect this to work. At the moment we are fixing every juju binary to open up a port to the world on a given port. As per spec I am worried this could be a mistake to leave our testing infrastructure potentially exposed in this way where we do have secrets to protect in these deployed binaries.

A few options are we only bind to localhost or we only make a unix socket file. Both of these options can be transported over an ssh connection to the client.